torchaudio/prototype/transducer.py

```diff
@@ -1,119 +1,289 @@
-def rnnt_loss(acts, labels, act_lens, label_lens, blank=0, reduction="mean"):
-    """Compute the RNN Transducer Loss.
-
-    The RNN Transducer loss (`Graves 2012 <https://arxiv.org/pdf/1211.3711.pdf>`__) extends the CTC loss by defining
-    a distribution over output sequences of all lengths, and by jointly modelling both input-output and output-output
-    dependencies.
-
-    The implementation uses `warp-transducer <https://github.com/HawkAaron/warp-transducer>`__.
-
-    Args:
-        acts (Tensor): Tensor of dimension (batch, time, label, class) containing output from network
-            before applying ``torch.nn.functional.log_softmax``.
-        labels (Tensor): Tensor of dimension (batch, max label length) containing the labels padded by zero
-        act_lens (Tensor): Tensor of dimension (batch) containing the length of each output sequence
-        label_lens (Tensor): Tensor of dimension (batch) containing the length of each output sequence
-        blank (int): blank label. (Default: ``0``)
-        reduction (string): If ``'sum'``, the output losses will be summed.
-            If ``'mean'``, the output losses will be divided by the target lengths and
-            then the mean over the batch is taken. If ``'none'``, no reduction will be applied.
-            (Default: ``'mean'``)
-    """
-
-    # NOTE manually done log_softmax for CPU version,
-    # log_softmax is computed within GPU version.
-    acts = torch.nn.functional.log_softmax(acts, -1)
-    return _RNNT.apply(acts, labels, act_lens, label_lens, blank, reduction)
+    def backward(ctx, output_gradients):
+        output_gradients = output_gradients.view(-1, 1, 1, 1).to(ctx.grads)
+        ctx.grads.mul_(output_gradients).to(ctx.grads)
 
+        return (
+            ctx.grads,  # logits
+            None,  # targets
+            None,  # src_lengths
+            None,  # tgt_lengths
+            None,  # blank
+            None,  # clamp
+            None,  # runtime_check
+            None,  # fused_log_smax
+            None,  # reuse_logits_for_grads
+        )
 
-@_mod_utils.requires_module("torchaudio._torchaudio")
-class RNNTLoss(Module):
-    """Compute the RNN Transducer Loss.
 
-    The RNN Transducer loss (`Graves 2012 <https://arxiv.org/pdf/1211.3711.pdf>`__) extends the CTC loss by defining
-    a distribution over output sequences of all lengths, and by jointly modelling both input-output and output-output
-    dependencies.
-
-    The implementation uses `warp-transducer <https://github.com/HawkAaron/warp-transducer>`__.
-
-    Args:
-        blank (int): blank label. (Default: ``0``)
-        reduction (string): If ``'sum'``, the output losses will be summed.
-            If ``'mean'``, the output losses will be divided by the target lengths and
-            then the mean over the batch is taken. If ``'none'``, no reduction will be applied.
-            (Default: ``'mean'``)
+class TransducerLoss(torch.nn.Module):
```

Comment:
Does this class also handle loss other than RNNT-Loss?
If not, can we name this more descriptive, like RNNTLoss? 

There are different kind of transducers. We cannot predict the future perfectly, but just in case we add other transducers, please name the class / functions more descriptive.